test/unit/test_mongodb_config.py

```diff
@@ -0,0 +1,106 @@
+import os
+import pytest
+from lib.Config import Configuration
+
+test_cases = [
+    pytest.param(
+        {
+            "username": "user",
+            "password": "pass",
+            "dns_srv": "True",
+            "expected_uri": "mongodb+srv://user:pass@localhost/testdb?authSource=admin&retryWrites=true&w=majority",
+            "expected_dbapi": "srv",
+        },
+        id="srv_with_auth",
+    ),
+    pytest.param(
+        {
+            "username": "user",
+            "password": "pass",
+            "dns_srv": "False",
+            "expected_uri": "mongodb://user:pass@localhost:27017/testdb?authSource=admin",
+            "expected_dbapi": None,
+        },
+        id="no_srv_with_auth",
+    ),
+    pytest.param(
+        {
+            "username": "",
+            "password": "",
+            "dns_srv": "True",
+            "expected_uri": "mongodb://localhost:27017/testdb",
+            "expected_dbapi": "srv",
```

Comment:
This assertion for `expected_uri` looks suspicious. While the modification aligns with the current logic (and the test would also have passed with the previous implementation of `getMongoUri()`), in reality it should be possible to use SRV records without authentication, shouldn’t it? If so, this case should likely have `"expected_uri": "mongodb+srv://localhost/testdb"`. Simplifying the function seems to reveal underlying logic errors, correct?

Reply:
I checked https://github.com/cve-search/CveXplore/pull/333 for this same logic error and CveXplore does it properly already.

While checking that I noticed that it adds the `user:pass@` to the URL even with empty values, generating `:@`. 

That is addressed in https://github.com/cve-search/CveXplore/pull/335.